kernel/kselftest.py

```diff
@@ -81,17 +81,10 @@ def setUp(self):
             location = self.params.get('location', default='https://github.c'
                                        'om/torvalds/linux/archive/master.zip')
             path = ''
-            git_location = True
-            kselfdir = ''
-            if ".zip" in location:
-                kselfdir = "kselftest.zip"
-                git_location = False
-            elif ".tar" in location:
-                kselfdir = "kselftest.tar"
-                git_location = False
-            if not git_location:
-                tarball = self.fetch_asset(kselfdir, locations=[location],
-                                           expire='1d')
+            match = next((x for x in [".zip", ".tar"] if x in location), None)
```

Comment:
and why for `x`  travis not a complaint  about pylint standard  let me check that  and it should be meaningful